libs/src/utils/multiProvider.js

```diff
@@ -0,0 +1,44 @@
+const { promisify, callbackify } = require('util')
+const Web3 = require('../web3')
+const { shuffle } = require('lodash')
+
+class MultiProvider extends Web3.providers.HttpProvider {
+  // web3 consumes a provider object on initialization
+  // ref: https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3/types/index.d.ts#L31
+  // which references: https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3-core/types/index.d.ts#L436
+  // MultiProivder implements AbstractProvider which can be consumed by web3
+
+  constructor (providers) {
+    super()
+
+    if (typeof providers === 'string') {
+      providers = providers.split(',')
+    } else if (!Array.isArray(providers)) {
+      providers = [providers]
+    }
+
+    providers = providers.map(provider => (new Web3(provider)).eth.currentProvider)
+
+    if (!providers.every(provider => provider.sendAsync || provider.send)) {
+      throw new Error('Some providers do not have a send method to use.')
+    }
+
+    this.providers = providers
+    this.send = callbackify(this._send.bind(this)) // web3 only supports callback functions and not async
+  }
+
+  _send (payload) {
```

Comment:
can you add comment explaining why we need to overwrite and what the overridden function does